prql-compiler/tests/integration-rdbms/connection.rs

```diff
@@ -0,0 +1,227 @@
+use std::time::SystemTime;
+
+use chrono::{DateTime, Utc};
+use mysql::prelude::Queryable;
+use mysql::Value;
+use pg_bigdecimal::PgNumeric;
+use postgres::types::Type;
+use tiberius::numeric::BigDecimal;
+use tiberius::*;
+use tokio::net::TcpStream;
+use tokio::runtime::Runtime;
+use tokio_util::compat::Compat;
+
+use prql_compiler::sql::Dialect;
+
+pub type Row = Vec<String>;
+
+pub struct DuckDBConnection(pub duckdb::Connection);
+
+pub struct SQLiteConnection(pub rusqlite::Connection);
+
+pub struct PostgresConnection(pub postgres::Client);
+
+pub struct MysqlConnection(pub mysql::Pool);
+
+pub struct MssqlConnection(pub tiberius::Client<Compat<TcpStream>>);
+
+pub trait DBConnection {
+    fn run_query(&mut self, sql: &str, runtime: &Runtime) -> Vec<Row>;
+
+    fn get_dialect(&self) -> Dialect;
+}
+
+impl DBConnection for DuckDBConnection {
+    fn run_query(&mut self, sql: &str, _runtime: &Runtime) -> Vec<Row> {
+        let mut statement = self.0.prepare(sql).unwrap();
+        let mut rows = statement.query([]).unwrap();
+        let mut vec = vec![];
+        while let Ok(Some(row)) = rows.next() {
+            let mut columns = vec![];
+            // row.len() always gives 1. hence this workaround
+            for i in 0.. {
+                let v_ref = match row.get_ref(i) {
+                    Ok(v) => v,
+                    Err(_) => {
+                        break;
+                    }
+                };
+                let value = match v_ref {
```

Comment:
BTW to the extent that `v_ref.to_string()` gets us there, that's fine!

Reply:
`ToString` is not implemented for `ValueRef`